refactor(test): improve DataAvailabilityChallenge test structure and documentation

aliersh opened the pull request:
- consolidate test initialization into a single DataAvailabilityChallenge_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- add @title and @notice tags to all test contracts
- ensure comment lines stay within 100 character limit

aliersh committed:
refactor(test): improve DataAvailabilityChallenge test structure and documentation

- consolidate test initialization into a single DataAvailabilityChallenge_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- add @title and @notice tags to all test contracts
- ensure comment lines stay within 100 character limit

aliersh reviewed (commented):
All the following functions are either not directly tested in this test file or are only covered indirectly through other tests:

initialize()
getChallenge()
getChallengeStatus()
computeCommitmentKeccak256()

smartcontracts commented on packages/contracts-bedrock/test/L1/DataAvailabilityChallenge.t.sol:
```diff
@@ -11,25 +11,92 @@ import { computeCommitmentKeccak256 } from "src/L1/DataAvailabilityChallenge.sol
+/// @title DataAvailabilityChallenge_SetBondSize_Test
+/// @notice Test contract for DataAvailabilityChallenge `setBondSize` function.
+contract DataAvailabilityChallenge_SetBondSize_Test is DataAvailabilityChallenge_TestInit {
+    function test_setBondSize_succeeds() public {
```
Can you add comments to these test functions?

aliersh replied:
done

aliersh committed:
refactor(test): add natspec comments to test functions

smartcontracts commented:
/ci authorize 4967fb5d241832914db847b563983b2e29d4cc41

aliersh committed:
refactor(test): fix formatting and enforce documentation line limits

- ran `just pre-pr` to resolve format check failures
- ensured all inline comments stay within the 100-character limit

smartcontracts commented:
/ci authorize e4d9c1df8792c0697e066c291ad9e4c936f9d782
